plugins/inputs/wireless/wireless_mac.go

```diff
@@ -0,0 +1,78 @@
+//go:build darwin
+// +build darwin
+
+package wireless
+
+import (
+	"errors"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"os/exec"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+// default executable path
+const (
+	OSXCMD = "/System/Library/PrivateFrameworks/Apple80211.framework/Versions/Current/Resources/airport -I"
+)
+
+func (w *Wireless) exe_cmd(cmd string, wg *sync.WaitGroup) ([]byte, error) {
+	parts := strings.Fields(cmd)
+	head := parts[0]
+	parts = parts[1:]
+	out, err := exec.Command(head, parts...).Output()
+	wg.Done() // Need to signal to waitgroup that this goroutine is done
+	return out, err
+}
+
+func (w *Wireless) Gather(acc telegraf.Accumulator) error {
+	if runtime.GOOS == "darwin" {
+		// collect MAC OS wireless data
+		wg := new(sync.WaitGroup)
+		wg.Add(3)
+		wireless, err := w.exe_cmd(OSXCMD, wg)
+		if err != nil {
+			return err
+		}
+		metrics, tags, err := w.loadMacWirelessTable(wireless, w.DumpZeros)
+		if err != nil {
+			return err
+		}
+		acc.AddGauge("wireless", metrics, tags)
+		return nil
+	}
+	return errors.New("OS Not Supported")
+}
+
+func (w *Wireless) loadMacWirelessTable(table []byte, dumpZeros bool) (map[string]interface{}, map[string]string, error) {
+	metrics := strings.Split(string(string(table)), "\n")
+	tags := make(map[string]string)
+	points := make(map[string]interface{})
+	for x := 0; x < len(metrics); x++ {
+		fm := strings.Split(strings.TrimSpace(metrics[x]), ":")
+		if len(fm) > 1 {
+			name := strings.Replace(strings.Trim(strings.TrimSpace(fm[0]), ":"), " ", "_", -1)
+			v := strings.TrimSpace(fm[1])
+			val, err := strconv.Atoi(v)
+			if err == nil { // it's a number
+				if !dumpZeros && val == 0 {
+					continue
+				}
+				points[name] = int64(val)
+			} else { // it's a string
+				tags[name] = strings.Replace(v, " ", "_", -1)
+			}
+		}
+	}
+	tags["interface"] = "airport"
+	return points, tags, nil
+}
+
+func init() {
+	inputs.Add("wireless", func() telegraf.Input {
+		return &Wireless{}
+	})
+}
```

Comment:
Not sure if this needs to be added?

Reply:
Does that answer my question?

Reply:
Well, it's present on the `wireless_linux` portion, and the `wireless_nosupport` portion so I figured it was required here. Am I wrong?

Reply:
Easy to check, remove it and see if the plugin / tests still work...

Reply:
From the [Docs](https://github.com/influxdata/telegraf/blob/master/docs/INPUTS.md) on building plugins 

> Input Plugins should call inputs.Add in their init function to register themselves. See below for a quick example.

Reply:
ALL  plugins usually only have one .go file where this is defined. This specific plugin seems to have it in the main file `wireless.go` but also in `wireless_mac.go` which seems redundant to me. Would you please quickly test, so we are sure?